boards/arm/arduino_giga/arduino_giga_m7_defconfig

```diff
@@ -0,0 +1,36 @@
+# Copyright 2023 Google LLC
+# SPDX-License-Identifier: Apache-2.0
+
+CONFIG_SOC_SERIES_STM32H7X=y
+CONFIG_SOC_STM32H747XX=y
+
+# Board config should be specified since there are 2 possible targets
+CONFIG_BOARD_ARDUINO_GIGA_M7=y
+
+# Disable the internal SMPS regulator
+CONFIG_POWER_SUPPLY_DIRECT_SMPS=n
+
+# Enable GPIO
+CONFIG_GPIO=y
+
+# Enable clocks
+CONFIG_CLOCK_CONTROL=y
+
+# Enable MPU
+CONFIG_ARM_MPU=y
+
+# Enable HW stack protection
+CONFIG_HW_STACK_PROTECTION=y
+
+# Enable uart driver
+CONFIG_SERIAL=y
+
+# Console
+CONFIG_CONSOLE=y
+CONFIG_UART_CONSOLE=y
+
+# Enable pin controller
+CONFIG_PINCTRL=y
+
+# Use zephyr,code-partition as flash offset
+CONFIG_USE_DT_CODE_PARTITION=y
```

Comment:
Is this always required? If someone just wants to build hello_world without a bootloader, this should be placed at 0x40000?

Reply:
Yeah the board comes with an Arduino bootloader that look for the application at 0x40000. I see how that is very confusing since it can also be used with the ROM bootloader which is going to not have the offset and wipe the factory one, and both support USB-DFU.

The intended use would be with the Arduino one since that's compatible with their IDE and all the other stuff, and that's what the `board.cmake` VID/PID points to, so this should stay, but I added a big note explaining the two bootloader story in the documentation page. Could not find a page on how to re-flash the factory bootloader but it should land there eventually.

Reply:
@tejlmand @mbolivar-nordic this presents an interesting conundrum with things like `west flash --recover` - what if a user has a bootloader (which is not theirs/outside the zephyr build system) and wants to flash their application to the board or set of applications and recover the device with exception to the bootloader? I.e. wants the storage partition cleared also to remove any saved state of previous application testing. Is this something we ever want to support?

Reply:
Are you thinking about having a `--recover` option outside of the `nrfjprog` target? Like to pad the settings area to nulls? That would not be a bad idea, a use case I bumped into: the bluetooth HID  (and other ble samples) by default stores the bond information in flash and have no way of deleting the bond. If one unpairs the device then can't pair any other until the settings are cleared, either `nrfjprog -e`, or hacking the BLE shell in the sample or hack the code to call `ble_unpair`.

Reply:
@nordicjm yeah, it's a nice-to-have, but I have no plans to do anything like that